src/openforms/submissions/tests/factories.py

```diff
@@ -140,6 +177,33 @@ class SubmissionStepFactory(factory.django.DjangoModelFactory):
     class Meta:
         model = SubmissionStep
 
+    @classmethod
+    def create_with_variables(
+        cls,
+        **kwargs,
+    ) -> SubmissionStep:
+        submission_step = cls.create(**kwargs)
+
+        form_variables = FormVariable.objects.filter(
+            form=submission_step.submission.form
+        )
+        step_data = kwargs.get("data", {})
+
+        for variable in form_variables:
+            try:
+                value = glom(step_data, variable.key)
+            except PathAccessError:
+                continue
+
+            SubmissionValueVariableFactory.create(
+                submission=submission_step.submission,
+                key=variable.key,
+                value=value,
+                form_variable=variable,
+            )
+
+        return submission_step
```

Comment:
Shouldn't we just do always do this and thus override the super `create`? or am I missing something?